src/pretix/base/models/event.py

```diff
@@ -804,12 +804,18 @@ def copy_data_from(self, other):
                 v.item = i
                 v.save(force_insert=True)
 
-        for imv in ItemMetaValue.objects.filter(item__event=other).prefetch_related('item', 'property'):
+        for imv in ItemMetaValue.objects.filter(item__event=other).prefetch_related('property'):
             imv.pk = None
-            imv.property = item_meta_properties_map[imv.property.pk]
+            imv.property = item_meta_properties_map[imv.property_id]
             imv.item = item_map[imv.item.pk]
             imv.save(force_insert=True)
 
+        for imv in ItemVariationMetaValue.objects.filter(variation__item__event=other).prefetch_related('property'):
+            imv.pk = None
+            imv.property = item_meta_properties_map[imv.property.pk]
```

Comment:
Sometimes it is `imv.property_id`, sometimes `imv.property.pk` – is there a difference? `property_id` reads more like knowing the internals whereas `property.pk` requires an object to be instantiated?

Reply:
Yes, there is a difference. property_id does not require the property object to be loaded from the database. We can save a SQL query here by using property_id and removing the prefetch related :+1: